src/com/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -288,9 +289,10 @@
     ;; confused if the database doesn't exist but we open and close a
     ;; connection without creating anything.
     (sql/with-connection write-db
-                         (scf-store/validate-database-version #(System/exit 1))
+                         (scf-store/validate-database-version
```

Comment:
On another note, maybe should move this validation to somewhere before we try starting the service? Like right after we process that section of the config? In the spirit of failing fast. That way we can also use the `cli-error` or `config-error` to do the exiting.